libraries/monitor/src/logger.js

```diff
@@ -139,7 +139,7 @@ class Logger {
     elideSecrets(fields);
 
     if (this.metadata) {
-      fields.meta = this.metadata;
+      Object.assign(fields, this.metadata, fields);
```

Comment:
doesn't this mean that we can't override metadata at a childMonitor level? Why the second `fields`?

Reply:
> doesn't this mean that we can't override metadata at a childMonitor level?

This is handled in the `childMonitor` function:
https://github.com/taskcluster/taskcluster/blob/d9ee68fdeb464e6f14eeea430c5a1dab4758f749/libraries/monitor/src/monitor.js#L93-L96

> Why the second `fields`?

The idea was to make fields override metadata, but it doesn't work, probably due to order of operations:
```
> let fields = {a: 10, b: 20};
undefined
> let metadata = {b: 300, c: 300};
undefined
> Object.assign(fields, metadata, fields);
{ a: 10, b: 300, c: 300 }
```
I'll just use `...`.